### PR TITLE
Only UpdatePrimitives and Draw tracks on screen

### PR DESCRIPTION
**Pull request by karupayun**
Optimization easy to make now that we are not setting
positions neither sizes while drawing anymore.

After calculation all the positions in WorldSpace we only
UpdatePrimitives and Draw to the tracks which have some part visible in
screen.

As a result, in captures where we have a lot of Tracks with timers
in several depths, we are much more faster at Updating Primitives. For
example, in the capture generated by Pierric using OrbitTest, with the
initial zoom I'm seeing UpdatePrimitives being 5/6 times faster, but
maybe could vary based on the resolution.

Anyway, in addition with https://github.com/google/orbit/pull/2780,
which is an optimization only for thread tracks, we are aiming to be
only bounded by the number of pixels where we draw timers multiplied by
a log of the number of elements.

Test: Loaded a capture, start a new one.

### CODE REVIEW

File: src/OrbitGl/TrackManager.cpp [hunk 1]
```diff
@@ -78,6 +78,20 @@ std::vector<FrameTrack*> TrackManager::GetFrameTracks() const {
   return tracks;
 }
 
+std::vector<Track*> TrackManager::GetTracksOnScreen() const {
+  std::vector<Track*> visible_tracks;
+  for (auto track : GetVisibleTracks()) {
+    auto track_top_y = track->GetPos()[1];
+    auto track_bottom_y = track_top_y + track->GetHeight();
+    auto screen_top_y = viewport_->GetScreenTopLeftInWorld()[1];
+    auto screen_bottom_y = screen_top_y + viewport_->GetVisibleWorldHeight();
```
Comment (pierricgimmig) [inline src/OrbitGl/TrackManager.cpp:87]: Don't use auto for simple types: https://google.github.io/styleguide/cppguide.html#Type_deduction

File: src/OrbitGl/TimeGraph.cpp [hunk 2]
```diff
@@ -821,7 +821,7 @@ void TimeGraph::DrawIncompleteDataIntervals(Batcher& batcher, PickingMode pickin
 
 void TimeGraph::DrawTracks(Batcher& batcher, TextRenderer& text_renderer,
                            const DrawContext& draw_context) {
-  for (auto& track : track_manager_->GetVisibleTracks()) {
+  for (auto& track : track_manager_->GetTracksOnScreen()) {
```
Comment (pierricgimmig) [inline src/OrbitGl/TimeGraph.cpp:824]: Don't use auto for simple types: https://google.github.io/styleguide/cppguide.html#Type_deduction

File: src/OrbitGl/TrackManager.cpp [hunk 3]
```diff
@@ -317,7 +331,7 @@ int TrackManager::FindMovingTrackIndex() {
 
 void TrackManager::UpdateTrackPrimitives(Batcher* batcher, uint64_t min_tick, uint64_t max_tick,
                                          PickingMode picking_mode) {
-  for (auto& track : visible_tracks_) {
+  for (auto& track : GetTracksOnScreen()) {
```
Comment (pierricgimmig) [inline src/OrbitGl/TrackManager.cpp:334]: Don't use auto for simple types: https://google.github.io/styleguide/cppguide.html#Type_deduction

**Review by pierricgimmig — APPROVED** [review]
Thanks for this Pablo! Oh, maybe add 

```
IMGUI_VAR_TO_TEXT(time_graph_->GetTrackManager()->GetTracksOnScreen().size());
```

To CaptureWindow::RenderImGuiDebugUI() ?

**Comment by karupayun** [comment]
> Thanks for this Pablo! Oh, maybe add
> 
> ```
> IMGUI_VAR_TO_TEXT(time_graph_->GetTrackManager()->GetTracksOnScreen().size());
> ```
> 
> To CaptureWindow::RenderImGuiDebugUI() ?

Done! Thanks for the suggestion :)

**Review by reichlfl — APPROVED** [review]
Nice, thanks!

**Pull request merged**